packages/cli/src/Queue.ts

```diff
@@ -60,51 +65,62 @@ export class Queue {
 					: getRedisStandardClient(Redis, clientConfig, (type + '(bull)') as RedisClientType),
 		});
 
-		this.jobQueue.on('global:progress', (jobId, progress: WebhookResponse) => {
-			this.activeExecutions.resolveResponsePromise(
-				progress.executionId,
-				decodeWebhookResponse(progress.response),
-			);
-		});
-	}
+		const instanceType = config.getEnv('generic.instanceType');
+		if (instanceType === 'main') {
```

Comment:
Now with multi-main we might get duplicate counts if there are multiple main instances listening to this, so this code should subescribe to the `leadershipChange` event and only fire if current instance is leader